### ISSUE TITLE
Global encoder appears to have no effect on the output of the TUS-REC 2025 checkpoint

### ISSUE THREAD

**Issue by alicranck** [ImFusionGmbH/DualTrack#7]
Hi guys, first of all wanted to say thanks for releasing the code and checkpoints! The staged configs made it straightforward to reproduce your results.

While profiling the model I ran into something I can't explain, and I wanted to check whether it's expected or whether I'm misusing the API.

With the released `dualtrack_ft_tus_rec_2025` checkpoint, replacing the global encoder's input with rolled frames, shuffled frames, pure Gaussian noise, or zeros leaves the model's output unchanged to within float32 rounding error.

I'm enclosing below a script to reproduce my experiment, that uses only this repo's code,
`configs/model/dualtrack_ft_tus_rec_2025.yaml` unmodified, and the released checkpoint.
The model is called through its documented signature `model(global_encoder_inputs, local_encoder_inputs)` - no library code is changed.
Run from repo root with:

```
python repro_global_encoder.py \
    --checkpoint data/checkpoints/dualtrack_ft_tus_rec_2025_v3_best.pt \
    --sweep <tus_rec_2025>/val/frames/050/RH_rotation.h5 --amp
```

[repro_global_encoder.py](https://github.com/user-attachments/files/30784379/repro_global_encoder.py)

The last line localises it upstream of the fusion module - `global_encoder` itself already returns essentially the same features 
for real keyframes and for pure noise, so the invariance source isn't the cross-attention.

Behaviour is also quite different on the 2024 checkpoint so could be something that the finetuning is doing?
Appreciate any input you have on this, Thanks!

**Comment by pfrwilson** [comment]
Hi @alicranck, thanks for the detailed report and the repro script! You're not misusing the API - we've confirmed the behaviour, and the global encoder's contribution in the `dualtrack_ft_tus_rec_2025` checkpoint is indeed degenerate. Predictions from that checkpoint are effectively driven by the local encoder and fusion module, which is why the reconstruction metrics didn't surface it.

We've since observed a training instability in isolated runs that we believe accounts for the collapse. With the local encoder compensating, it went unnoticed during the 2025 fine-tuning. This is specific to those runs; the 2024 checkpoint (`dualtrack_final.pt`) is unaffected, consistent with what you observed. Improving the stability of that stage is on our list, and we'll leave this issue open to track it.

In the meantime, if you're fine-tuning on your own data and want a functioning global branch, we'd suggest initializing from `dualtrack_final.pt` rather than the 2025 checkpoint.

Thanks again for taking the time to isolate this so precisely. We really appreciate your report and your interest in the project.
